docs/scos/dev/front-end-development/202212.0/oryx/i18n/localization.md

```diff
@@ -0,0 +1,138 @@
+---
+title: "Oryx: Localization"
+description: Localizations are decoupled from component implementations
+template: concept-topic-template
+last_updated: July 23, 2023
+---
+
+Localization is an important part of internationalization (`i18n`) and is concerned with creating variations of the same application in different locales.
+
+While most localized content is managed in other systems, like product or content management systems, components often use small text fragments in the UI. Examples are static site labels or aria labels
+
+To ensure that all components can be customized and localized, Oryx does not provide _any_ hardcoded text. Instead, components use _tokens_ that function as a reference to a text. When there's not (localized) text available for such reference, Oryx renders the token in a human readable format.
+
+## Translation keys
+
+Translation keys, also knows as "i18n tokens", are used to resolve a localized text. The tokens are created with the following conventions in mind:
+
+- tokens are written in english
+- tokens are written in kebab-case format, e.g. `my-token`
+- tokens are organized by domains, e.g. `cart.add-to-cart`
+- tokens support context variables, which are added inside angle brackets using with camelCase format, e.g. `cart.totals.<count>-items`
+
+## Resolve translations from translation resources
+
+The localization of labels is driven by the current language and the translation key. Translations are supposed to be provided as additional lazy loaded resources next to the component implementation, although they can be added as part of the static resources that are loaded in Oryx as well. If the resources are loaded as external resource, they can be provided by an API (e.g. 3rd party service) or by static JSON files.
+
+Oryx uses the "current" language to "lookup" the available labels. When the language is `en`, the locales for English are being resolved.
+
+If a language resource can be resolved, the translation key is evaluated against the available translations in the resource, using all "parts" of the key. For example, if you have a translation key that contains multiple parts (e.g. `cart.increase`), you can provide a global translation for just the `increase` part or provide a translation for `cart.increase`. This mechanism allows for a single global translation of `decrease` that might affect multiple components throughout multiple components. This results in a consistent and convenient translation mechanism, while remaining flexible to add very specific translations for some components.

```

Comment:
We were talking about `cart.increase` so why there is an example with `decrease`?
```suggestion
If a language resource can be resolved, the translation key is evaluated against the available translations in the resource, using all "parts" of the key. For example, if you have a translation key that contains multiple parts (e.g. `cart.increase`), you can provide a global translation for just the `increase` part or provide a translation for `cart.increase`. This mechanism allows for a single global translation of `increase` that might affect multiple components throughout multiple components. This results in a consistent and convenient translation mechanism, while remaining flexible to add very specific translations for some components.

```